src/main/content/_assets/js/guide-multipane.js

```diff
@@ -334,29 +345,29 @@ $(document).ready(function() {
             // Switch to the correct tab
             var tab = code_sections[header.id][fileIndex].tab;
             setActiveTab(tab);                   
-            showCorrectCodeBlock(header.id, fileIndex, false);            
+            showCorrectCodeBlock(header.id, fileIndex, false);
 
             // Highlight the code
-            if(highlightCode){
-                var ranges = hotspot.data('highlight-ranges');
+            var ranges = hotspot.data('highlight-ranges');
+            if(ranges){
                 ranges = ranges.split(',');
                 for(var i = 0; i < ranges.length; i++){
                     var lines = ranges[i].split('-');
                     if(lines.length === 2){
                         var fromLine = parseInt(lines[0]);
                         var toLine = parseInt(lines[1]);
                         var num_Lines = parseInt(code_block.find('.line-numbers').last().text());
-                        if(fromLine && toLine){              
+                        if(fromLine && toLine){
                             // If a hotspot refers to a whole file, do not highlight it.    
-                            if(!(fromLine === 1 && toLine === num_Lines)){               
+                            if(!(fromLine === 1 && toLine >= num_Lines)){               
```

Comment:
We discussed offline about `toLine >= numLines`.

I found it odd that the guide author would annotate their `adoc` file with a line range greater than the file's max number of lines.  It was discovered that guide authors used a line range of `1 - max line` as a way make the view focus on the code file, if the code file is in a background tab.  We also avoided highlighting the whole file even though the hotspot range is 1 through the end of the file.  Hotspot with range should mean highlight lines, even if they put a range of 1 through file's max lines.

Given that we are introducing `[hotspot]` with no range to indicate to bring the whole file into view if it is in a background tab AND not highlight any lines, this `if(!(fromLine === 1 && toLine >= num_Lines)){ ` check will no longer be needed.

We also now have a new `if(ranges)` guard above, so the only time we use this code is when the author wants to apply highlighting to the code lines via a hotspot with line range (ie `hotspot=3-60`).